src/services/array-functions.js

```diff
@@ -21,15 +21,30 @@ export function map(theArray, fnc){
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
 export function filter(theArray, fnc){
+  let myArray = [];
+  for(let i=0; i< theArray.lenght; i ++){
+  if (fnc(theArray[i])) {
+    myArray.push(theArray[i]);
+  }
+  }
+  return myArray;
+  }
 
 }
 
-
+a
 //loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true return the item
 //return null
 export function find(theArray, fnc){
+  let item = null;
+for(let i=0; i<theArray.length; i++){
+  if(fnc(theArray[i])) {
+    item = theArray[i];
```

Comment:
This approach is okay, but you should really just return the item from the array immediately. There is no point to continue looping when you already found the thing you are looking for.